Player profile dev

jeremy900425 opened the pull request:
# 完成player-profile spec中的功能

## Checklist

| Item          | Is checked? | Description                                        |
| ------------- | ----------- | -------------------------------------------------- |
| Explanation   | V          | My changes are well-explained.                     |
| Documentation | V           | My works are well-commented or clear.              |
| Tests         | X           | I have added tests or reports to cover my changes. |

jeremy900425 committed:
test branch is create

jeremy900425 committed:
get body_json data

jeremy900425 committed:
complete all player profile function remain SQL process

jeremy900425 committed:
complete all Player-profile function

jeremy900425 committed:
reslove conflict

jeremy900425 committed:
重構state功能

AsherJingkongChen reviewed (commented):
Reviewing

AsherJingkongChen reviewed (changes requested):
Reviewing *

AsherJingkongChen commented on app/routes.py:
```diff
@@ -173,48 +173,92 @@ def GET_teams():
     sort_field = request.args.get('sort_field')
     sort_order = request.args.get('sort_order')
 
-    if page_offset is None or page_length is None or not sort_field or not sort_order:
```
## Warn

Wrongly removing

AsherJingkongChen commented on app/routes.py:
```diff
@@ -173,48 +173,92 @@ def GET_teams():
+    if 'name' not in data:
+        return jsonify({"message": "Your request is invalid."}), 400
 
-@bp_web_api.route('/teams/<int:id>', methods=['PATCH'])
```
## Warn

Wrongly removing

AsherJingkongChen commented on app/dataProcess_player_profiles.py:
```diff
@@ -1,72 +1,216 @@
+        Team t ON p.TID = t.TID
+    """
+    # 添加排序條件
+    if sort_field in ['Player', 'Team', 'Position', 'Birthdate', 'Height', 'Weight', 'Country', 'Fname']:
```
## Small warn

Need sort field mapping

AsherJingkongChen replied:
I may solve it by myself. Though it is about validation.

AsherJingkongChen reviewed (changes requested):
Roughly reviewed. Not detailed.

AsherJingkongChen committed:
(patch): [Warn] Wrongly removing

(sub-module): Remove doc and update web
- doc/ is already removed in `main` branch
